src/Payment/Service.php

```diff
@@ -28,7 +30,27 @@ public function create(
         string $currency = 'GBP',
         string $description = null,
         array $metadata = [],
-        string $customerUid = null
+        string $customerUid = null,
+        array $params = []
+    ): Payment;
+
+    /**
+     * Tries to reserve funds.
+     *
+     * @param Token $token
+     * @param int $amount
+     * @param string $currency
+     * @param array $params
+     *
+     * @return Payment
+     *
+     * @throws Exception|\Stripe\Error\Api
+     */
+    public function reserve(
+        Token $token,
+        int $amount,
+        string $currency = 'GBP',
+        array $params = []
```

Comment:
is there a reason why we've gone to a params array here rather than known/specific ​arguments